sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -128,6 +128,54 @@ public static void setApiUrl(String apiUrl) {
         return getInstance().setInstanceEnabledAsync(enabled);
     }
 
+    /**
+     * Check whether offline simulation is enabled or not.
+     *
+     * @return result being <code>true</code> if enabled, <code>false</code> otherwise.
+     * @see AppCenterFuture
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"}) // TODO Remove warning suppress after release.
+    public static boolean isSimulateOffline() {
+        return getInstance().isSimulateOfflineInstance();
+    }
+
+    /**
+     * Enable or disable offline simulation.
+     *
+     * @param offline <code>true</code> to simulate device being offline, <code>false</code> to go back to the original network state of the device.
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"}) // TODO Remove warning suppress after release.
+    public static void setSimulateOffline(boolean offline) {
+        getInstance().setSimulateOfflineInstance(offline);
+    }
+
+    /**
+     * Check whether offline simulation is enabled or not.
+     *
+     * @return result being <code>true</code> if enabled, <code>false</code> otherwise.
+     * @see AppCenterFuture
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"}) // TODO Remove warning suppress after release.
+    public boolean isSimulateOfflineInstance() {
+        if (mHttpClient != null) {
+            return mHttpClient.isSimulateOffline();
+        }
+
+        return false;
+    }
+
+    /**
+     * Enable or disable offline simulation.
+     *
+     * @param offline <code>true</code> to simulate device being offline, <code>false</code> to go back to the original network state of the device.
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"}) // TODO Remove warning suppress after release.
+    public void setSimulateOfflineInstance(boolean offline) {
+        if (mHttpClient != null) {
+            mHttpClient.setSimulateOffline(offline);
```

Comment:
so we don't resume operations when coming back online?
this is different than what network state handler is doing that will resume calls when network back. I think we need to talk more about this feature.